torch/testing/_internal/common_utils.py

```diff
@@ -1426,111 +1339,37 @@ def _is_dict(self, obj):
     # TODO: default exact_device to True
     def assertEqual(self, x, y, msg: Optional[str] = None, *,
                     atol: Optional[float] = None, rtol: Optional[float] = None,
-                    equal_nan=True, exact_dtype=True, exact_device=False) -> None:
+                    equal_nan=True, exact_dtype=True, exact_device=False, exact_stride=False,
```

Comment:
Just so I'm sure I understand this change, this will still be a more restrictive version of assertEqual because of the type checking we now do, even with `allow_subclasses=True`?

Reply:
Yes, although this is relaxed for `numpy.ndarray`'s. So if someone wants to check another unrelated tensor-like and a tensor this will fail. There was no such case in the whole test suite.